Tools/CISupport/ews-build/steps.py

```diff
@@ -6874,3 +6905,330 @@ def doStepIf(self, step):
+        if self.getProperty('unexpected_failing_files', 0):
+            return defer.returnValue(FAILURE)
+        return defer.returnValue(SUCCESS)
+
+    def loadResultsData(self, unexpected_results_path):
+        with open(unexpected_results_path) as f:
+            unexpected_results_data = json.load(f)
+        return unexpected_results_data
+
+    @defer.inlineCallbacks
+    def getFilesPerProject(self, unexpected_results_data, type):
+        total_file_list = []
+        is_log = 0
+        for project, data in unexpected_results_data[type].items():
+            log_content = ''
+            for checker, files in data.items():
+                if files:
+                    total_file_list.extend(files)
+                    file_str = '\n'.join(files)
+                    log_content += f'=> {checker}\n\n{file_str}\n\n'
+            if log_content:
+                yield self._addToLog(f'{project}-unexpected-{type}', log_content)
+                is_log += 1
+        self.setProperty(f'{type}', total_file_list)
+        return defer.returnValue(is_log)
+
+    def doStepIf(self, step):
+        return self.getProperty('unexpected_failing_files', 0) or self.getProperty('unexpected_passing_files', 0) or self.getProperty('unexpected_new_issues', 0)
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+    def resultDirectoryURL(self):
+        path = self.resultDirectory.replace('public_html/results/', '') + '/'
+        return f'{S3_RESULTS_URL}{path}'
+
+    def getResultSummary(self):
+        num_failures = self.getProperty('unexpected_failing_files', 0)
+        num_passes = self.getProperty('unexpected_passing_files', 0)
+        num_issues = self.getProperty('unexpected_new_issues', 0)
+        failing_files = (", ").join(self.getProperty('failures', []))
+        passing_files = (", ").join(self.getProperty('passes', []))
+        results_summary = ''
+
+        results_link = self.resultDirectoryURL() + SCAN_BUILD_OUTPUT_DIR + "/new-results.html"
+        build_link = f'{self.master.config.buildbotURL}#/builders/{self.build._builderid}/builds/{self.build.number}'
+        formatted_build_link = f'[#{self.getProperty("buildnumber", "")}]({build_link})'
+
+        if num_failures:
+            pluralSuffix = 's' if num_issues > 1 else ''
+            comment = f"Smart Pointer Build {formatted_build_link}: Found [{num_issues} new failure{pluralSuffix}]({results_link}), blocking PR #{self.getProperty('github.number')}.\n"
+            comment += 'Please address these issues before landing. See [WebKit Guidelines for Safer C++ Programming](https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines).\n(cc @rniwa)'
+            self.setProperty('comment_text', comment)
+            self.build.addStepsAfterCurrentStep([LeaveComment(), SetCommitQueueMinusFlagOnPatch(), BlockPullRequest()])
+            results_summary = f'Found {num_issues} new failure{pluralSuffix} in {failing_files}'
```

Comment:
What if there are thousands of issues found, this summary would be extremely long containing thousands of file-names, and the UI would be extremely messy. Same thing for `passing_files`.

To handle that, while generating `failing_files` above, truncate it to a set limit (e.g.: 50). e.g.: `failing_files = (", ").join(self.getProperty('failures', [])[:50])`

e.g.: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L3093